### PR TITLE
Added a progress bar for upload and download of object to a bucket.

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
This PR adds a progress bar to show the progress of upload and download of an object to bucket.

**Test Report Added?**:
/kind TESTED

**Test Report**:
**Upload Progress**

upload-progress

**Download Progress**
download-progress

**Special notes for your reviewer**:

### CODE REVIEW

File: src/app/business/block/bucket-detail/bucket-detail.component.ts [hunk 1]
```diff
@@ -151,42 +154,42 @@ export class BucketDetailComponent implements OnInit {
...
-        
-        this.BucketService.copyObject(this.bucketId + '/' + key, '', options).subscribe((res) => {
-          this.isReadyPast = true;
-          window.sessionStorage['searchIndex'] = "";
-          this.getAlldir();
-          res
-        }, (error)=>{
-          window.sessionStorage['searchIndex'] = "";
-        });
+          let url = "/" + this.bucketId + '/' + key;
```
Comment (kumarashit) [inline src/app/business/block/bucket-detail/bucket-detail.component.ts:157]: This indentation does not look correct

Comment (kumarashit) [reply src/app/business/block/bucket-detail/bucket-detail.component.ts:157]: Ok...U may have to indent 156 line for four spaces

**Comment by kumarashit** [comment]
There is already a functionality in the backend to report the progress value. I remember C2C had put this. Why don't we leverage that and show in-line than showing it at bottom

**Comment by anvithks** [comment]
@kumarashit What C2C implemented was progress of Migration tasks. This was through the API and the progress is part of the API response of the Jobs API.
What I have implemented here is showing the upload and download progress on client side. 
I am using the PrimeNG ProgressBar module and have leveraged existing functionality provided by the `xhr.onprogress` , `xhr.upload.progress` methods.
The Upload and Download notification is existing and I have just added the progress bar to the same location. The design of the notification window can be changed. 
We can look at it at a later date.

**Comment by kumarashit** [comment]
> @kumarashit What C2C implemented was progress of Migration tasks. This was through the API and the progress is part of the API response of the Jobs API.
> What I have implemented here is showing the upload and download progress on client side.
> I am using the PrimeNG ProgressBar module and have leveraged existing functionality provided by the `xhr.onprogress` , `xhr.upload.progress` methods.
> The Upload and Download notification is existing and I have just added the progress bar to the same location. The design of the notification window can be changed.
> We can look at it at a later date.

OK. Thanks

**Pull request merged**